provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -1798,6 +1812,108 @@ func generateSchema() schema.PackageSpec {
 					Required: []string{"content", "contentType"},
 				},
 			},
+			"eks:index:ResolveConflictsOnUpdate": {
+				ObjectTypeSpec: schema.ObjectTypeSpec{
+					Type: "string",
+					Description: "How to resolve field value conflicts for an Amazon EKS add-on if you've changed a value from the Amazon EKS default value. " +
+						"Valid values are `NONE`, `OVERWRITE`, and `PRESERVE`. For more details see the [UpdateAddon](https://docs.aws.amazon.com/eks/latest/APIReference/API_UpdateAddon.html) API Docs.",
+				},
+				Enum: []schema.EnumValueSpec{
+					{
+						Name:        "None",
+						Value:       "NONE",
+						Description: "Amazon EKS doesn't change the value. The update might fail.",
+					},
+					{
+						Name:        "Overwrite",
+						Value:       "OVERWRITE",
+						Description: "Amazon EKS overwrites the changed value back to the Amazon EKS default value.",
+					},
+					{
+						Name:        "Preserve",
+						Value:       "PRESERVE",
+						Description: "Amazon EKS preserves the value. If you choose this option, we recommend that you test any field and value changes on a non-production cluster before updating the add-on on your production cluster.",
+					},
+				},
+			},
+			"eks:index:ResolveConflictsOnCreate": {
+				ObjectTypeSpec: schema.ObjectTypeSpec{
+					Type: "string",
+					Description: "How to resolve field value conflicts when migrating a self-managed add-on to an Amazon EKS add-on. " +
+						"Valid values are `NONE` and `OVERWRITE`. For more details see the [CreateAddon](https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateAddon.html) API Docs.",
+				},
+				Enum: []schema.EnumValueSpec{
+					{
+						Name:        "None",
+						Value:       "NONE",
+						Description: "If the self-managed version of the add-on is installed on your cluster, Amazon EKS doesn't change the value. Creation of the add-on might fail.",
+					},
+					{
+						Name:  "Overwrite",
+						Value: "OVERWRITE",
+						Description: "If the self-managed version of the add-on is installed on your cluster and the Amazon EKS default value is different than the existing value, " +
+							"Amazon EKS changes the value to the Amazon EKS default value.",
+					},
+				},
+			},
+			"eks:index:CoreDnsAddonOptions": {
+				ObjectTypeSpec: schema.ObjectTypeSpec{
+					Type: "object",
+					Properties: map[string]schema.PropertySpec{
+						"enabled": {
```

Comment:
The Go design would be `disabled` with default: false, so missing and false are the same. But perhaps we intentionally prefer the extra "enabled: true" marker.

Reply:
Those resources are deployed by default on EKS

Reply:
Most people will be using it in TypeScript 😀

Reply:
I'm thinking Go has good style in here :) But deferring to y'all.

Reply:
I just had a look through other settings in the provider and more are called `enable...` then `disable...`. Many of those are rooted in AWS settings, so I'd vote for `enabled` here to stay more consistent with what the provider is already doing

Reply:
Yeah if this is what AWS users expect it's perfect.